geode-core/build.gradle

```diff
@@ -259,29 +259,13 @@ dependencies {
   implementation('net.sf.jopt-simple:jopt-simple')
 
   //Log4j is used everywhere
-  implementation('org.apache.logging.log4j:log4j-api')
-  implementation('org.apache.logging.log4j:log4j-core') {
-    ext.optional = true
-  }
+  api('org.apache.logging.log4j:log4j-api')
```

Comment:
We used `implementation` here before which would have hidden the log4j API from consumers of geode-core.  Is that not what we want in this new model, since we are now using `api`?

Reply:
LogService is an internal API used by modules other than geode-core. In a follow-up PR, I will need to move LogService to a User API package. It returns log4j-api Logger and is an API.

Reply:
> In a follow-up PR, I will need to move LogService to a User API package

Could you elaborate on this some more?  Why are we exposing LogService as User API?  Do we want users of geode to be able to write to geode logs?

Reply:
I've discussed this previously with @pivotal-jbarrett and others. I currently feel strongly that we should NOT export internal APIs between modules. I also feel strongly against defining SPIs within internal packages. I know that @upthewaterspout prefers internal SPIs and internal APIs, but I haven't heard any arguments for this that have swayed me yet.

To summarize: If all geode-* modules are going to get Loggers from LogService then I want LogService to live in `org.apache.geode.logging` package instead of an internal package. I believe this is more correct and within this PR, the class only exposes two public methods:
```
public static Logger getLogger();

public static Logger getLogger(String name);
```
I defined geode-core as having `api` dependency on log4j-api because LogService currently lives in geode-core and is returning Loggers from log4j-api. This portion of the PR was done while pairing with @rhoughton-pivot who helped me understand how to use `api` vs `implementation` in gradle.

I have no problem moving LogService to geode-logging instead of geode-core.

Reply:
@kirklund - I do prefer internal APIs for things that we only want to be used internally. I think that is the case for LogService - we don't intend for users to ever call this class, so it should not be public.

By making log4j2 an API dependency we make life worse for our users because we are pollute their compile time classpath with log4j2. How we do logging inside geode really shouldn't leak out to our users. Adding public API that depends on log4j2-api would commit us into to providing a log4j2 API to our users in the future, even if we switched over to using logback or something else internally.